asset-transfer-basic/chaincode-java/src/main/java/org/hyperledger/fabric/samples/assetTransfer/Asset.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.samples.assettransfer;
+
+import java.util.Objects;
+
+import org.hyperledger.fabric.contract.annotation.DataType;
+import org.hyperledger.fabric.contract.annotation.Property;
+
+import com.owlike.genson.annotation.JsonProperty;
+
+@DataType()
+public final class Asset {
+
+    @Property()
+    private final String assetID;
+
+    @Property()
+    private final String color;
+
+    @Property()
+    private final int size;
+
+    @Property()
+    private final String owner;
+
+    @Property()
+    private final int appraisedValue;
+
+    public String getAssetID() {
+        return assetID;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public String getOwner() {
+        return owner;
+    }
+
+    public int getAppraisedValue() {
+        return appraisedValue;
+    }
+
+    public Asset(@JsonProperty("assetID") final String assetID, @JsonProperty("color") final String color,
```

Comment:
To keep parity with Go chaincode, I am leaning to have exact same api as Go chaincode, so application-* of this basic sample will work with chaincode-* 

suggestion
@JsonProperty("ID")

since that is the json field in go: https://github.com/hyperledger/fabric-samples/blob/971af50cdfdffac9b7645db244d1c57474709ffd/asset-transfer-basic/chaincode-go/chaincode/smartcontract.go#L17

Reply:
I tried to use ID but this caused a strange issue. Storing assets was no problem. But when I would retrieve the asset, the key changed from ID -> iD. Changing it to assetID resolved this issue. I'm not sure why it had an issue with ID.

Reply:
Oh I forgot to mention, when retrieving the asset, it would change the key from ID -> iD, and it would also null the value: 
<img width="852" alt="Screen Shot 2020-07-24 at 10 35 50 AM" src="https://user-images.githubusercontent.com/67290864/88402669-7384e100-cd99-11ea-9002-65f95db0e1e0.png">

Reply:
so the json marshalling is defaulting to **camelCase**

it is worth trying explicit annotation on Asset properties 
```
@Property()
@JsonProperty("ID")
private final String assetID;
```

I haven't used genson. If above does not work, this lib may be a better option
```
  <groupId>com.fasterxml.jackson.core</groupId>
  <artifactId>jackson-annotations</artifactId>
```

https://github.com/FasterXML/jackson-annotations/wiki/Jackson-Annotations

Reply:
The explicit annotation results in duplicate key-value pairs. I'll look into the Jackson Annotations.

Reply:
After spending some time on this, Jackson serialization produces duplicate key-value pairs along with ignoring the fields. Both Jackson and Genson don't play nice with fields beginning with an upper-case. I'm going to leave the fields as camelCase. This can be modified later.